src/loggers/database-logger.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * A simple logger to allow query and memory usage logging.
+ *
+ * @package Yoast\WP\Free\Initializers
+ */
+
+namespace Yoast\WP\Free\Loggers;
+
+use Yoast\WP\Free\Conditionals\Development_Conditional;
+use Yoast\WP\Free\Integrations\Integration_Interface;
+use YoastSEO_Vendor\ORM;
+
+/**
+ * Class Database_Logger
+ */
+class Database_Logger implements Integration_Interface {
+	/**
+	 * @inheritDoc
+	 */
+	public function register_hooks() {
+		ORM::configure( 'logging', true );
+		ORM::configure( 'logger', [ $this, 'logger' ] );
+
+		add_action( 'shutdown', [ $this, 'log_output' ] );
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public static function get_conditionals() {
+		return [ Development_Conditional::class ];
+	}
+
+	/**
+	 * Private array of queries used for logging.
+	 *
+	 * @var array
+	 */
+	protected $query_log = [];
+
+	/**
+	 * Logs the query to a local variable for output on shutdown.
+	 *
+	 * @param string $query The query.
+	 * @param float  $time  The time the query took to execute.
+	 */
+	public function logger( $query, $time ) {
+		$query             = [
+			'query' => $query,
+			'time'  => $time
+		];
+		$this->query_log[] = $query;
+	}
+
+	/**
+	 * Outputs some logging.
+	 */
+	public function log_output() {
+		if (
+			( defined( 'DOING_AJAX' ) && DOING_AJAX ) ||
```

Comment:
For the AJAX and REST_REQUEST there are two wrapper methods in WordPress `wp_doing_ajax()` and `wp_doing_rest_request()`

Reply:
Did you just make up `wp_doing_rest_request`? :)

Reply:
My bad, it was a function where I made a patch for, but it was closed because someone else was working on a solution for that. 😬